exonum-java-binding-core/rust/src/utils/jni_cache.rs

```diff
@@ -31,128 +31,157 @@ pub extern "system" fn JNI_OnLoad(vm: JavaVM, _: *mut c_void) -> jint {
-pub fn get_class_get_name() -> JMethodID<'static> {
-    unsafe { CLASS_CACHED_GET_NAME.unwrap() }
+    /// Returns cached `JMethodID` for `UserTransactionAdapter.isValid()`.
+    pub fn verify_id() -> JMethodID<'static> {
+        unsafe { TRANSACTION_ADAPTER_VERIFY.unwrap() }
+    }
 }
 
-/// Returns cached `JMethodID` for `java.lang.Throwable.getMessage()`
-pub fn get_throwable_get_message() -> JMethodID<'static> {
-    unsafe { THROWABLE_CACHED_GET_MESSAGE.unwrap() }
-}
+/// Refers to the cached methods of the `UserServiceAdapter` class.
+pub mod service_adapter {
+    use super::*;
 
-/// Returns cached `JMethodID` for `UserTransactionAdapter.execute()`
-pub fn get_uta_execute() -> JMethodID<'static> {
-    unsafe { UTA_CACHED_EXECUTE.unwrap() }
+    /// Returns cached `JMethodID` for `UserServiceAdapter.getStateHashes()`.
+    pub fn state_hashes_id() -> JMethodID<'static> {
+        unsafe { SERVICE_ADAPTER_STATE_HASHES.unwrap() }
+    }
+
+    /// Returns cached `JMethodID` for `UserServiceAdapter.convertTransaction()`.
+    pub fn convert_transaction_id() -> JMethodID<'static> {
+        unsafe { SERVICE_ADAPTER_CONVERT_TRANSACTION.unwrap() }
+    }
 }
 
-/// Returns cached `JMethodID` for `UserTransactionAdapter.info()`
-pub fn get_uta_info() -> JMethodID<'static> {
-    unsafe { UTA_CACHED_INFO.unwrap() }
+/// Refers to the cached methods of the `java.lang.Object` class.
+pub mod object {
+    use super::*;
+
+    /// Returns cached `JMethodID` for `java.lang.Object.getClass()`.
+    pub fn get_class_id() -> JMethodID<'static> {
+        unsafe { OBJECT_GET_CLASS.unwrap() }
+    }
 }
 
-/// Returns cached `JMethodID` for `UserTransactionAdapter.isValid()`
-pub fn get_uta_verify() -> JMethodID<'static> {
-    unsafe { UTA_CACHED_VERIFY.unwrap() }
+/// Refers to the cached methods of the `java.lang.Class` class.
+pub mod class {
+    use super::*;
+
+    /// Returns cached `JMethodID` for `java.lang.Class.getName()`.
+    pub fn get_name_id() -> JMethodID<'static> {
+        unsafe { CLASS_GET_NAME.unwrap() }
+    }
 }
 
-/// Returns cached `JMethodID` for `UserServiceAdapter.getStateHashes()`
-pub fn get_usa_state_hashes() -> JMethodID<'static> {
-    unsafe { USA_CACHED_STATE_HASHES.unwrap() }
+/// Refers to the cached methods of the `java.lang.Throwable` class.
+pub mod throwable {
+    use super::*;
+
+    /// Returns cached `JMethodID` for `java.lang.Throwable.getMessage()`.
+    pub fn get_message_id() -> JMethodID<'static> {
+        unsafe { THROWABLE_GET_MESSAGE.unwrap() }
+    }
 }
 
-/// Returns cached `JMethodID` for `UserServiceAdapter.convertTransaction()`
-pub fn get_usa_convert_tx() -> JMethodID<'static> {
-    unsafe { USA_CACHED_CONVERT_TRANSACTION.unwrap() }
+/// Provides access to various cached classes.
+pub mod classes_refs {
+    use super::*;
+
+    /// Returns cached `JClass` for `java/lang/Error` as a `GlobalRef`.
+    pub fn java_lang_error() -> GlobalRef {
+        unsafe { JAVA_LANG_ERROR.clone().unwrap() }
```

Comment:
1. Why is it `unsafe`?

2. Do we clone an Option first, and then unwrap? Shan't we do the opposite?

Reply:
1. Accessing mutable static variables is unsafe
2. The order doesn't matter because cloning `None` doesn't involve any computation.

Reply:
The order might not matter in terms of performance, but in terms of readability it raises an unnecessary question why you clone Option.

Reply:
That makes sense, thanks.